File: my-folder/0049-group-anagrams/solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        
        n = len(strs)
        table = defaultdict()
        output = []

        for st in strs:
            m = "".join(sorted(st))
            if m in table:
                output[table[m]].append(st)
            else:
                table[m] = len(output)
                output.append([st])
        
        return output


        
        

File: my-folder/0049-group-anagrams/solution.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> table;
        for (auto str: strs){
            string str_new = str;
            sort(str_new.begin(), str_new.end());
            table[str_new].push_back(str);
        }
        vector<vector<string>> output;
        for (auto& [key,value]: table){
            output.push_back(value);
        }
        return output;
        
    }
};
